### PR TITLE
Setup CI to build and test

### PR DESCRIPTION
**Pull request by bradbeck**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+
```
Comment (rgreinho) [inline .github/workflows/go.yml]: Can we add caching?


```suggestion

```
        with:
       - uses: actions/cache@v3
          path: ~/go/pkg/mod
          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
          restore-keys: |
            ${{ runner.os }}-go-
      - name: Setup the project
        run: go mod download

**Review by rgreinho — CHANGES_REQUESTED** [review]
Not a blocker, but it would be nice to have I think.

File: .github/workflows/go.yml (hunk 1 above)
Comment (rgreinho) [reply .github/workflows/go.yml]: @bradbeck disagreed?

Comment (bradbeck) [reply .github/workflows/go.yml]: Oh, I just committed the suggested change, but I guess it wasn't formatted correctly. Will do that in a minute.

**Review by rgreinho — DISMISSED** [review]

**Review by rgreinho — APPROVED** [review]

**Review by sudo-bmitch — APPROVED** [review]

**Pull request merged**